src/SDK/MonitoringApplianceEmulator/ViewModels/AlertDetailsControlViewModel.cs

```diff
@@ -143,5 +148,115 @@ private set
+                            {
+                                tableValue = response[0].SelectToken(tableRef.ReferencePath) as JArray;
+                            }
+                            else
+                            {
+                                tableValue = (new JArray(response)).SelectToken(tableRef.ReferencePath) as JArray;
+                            }
+
+                            List<Dictionary<string, JToken>> values = tableValue
+                                .OfType<IDictionary<string, JToken>>()
+                                .Select(value => value.ToDictionary(item => item.Key, item => item.Value))
+                                .ToList();
+                            TableAlertProperty<Dictionary<string, JToken>> displayTable = new TableAlertProperty<Dictionary<string, JToken>>(tableRef.PropertyName, tableRef.DisplayName, tableRef.Order, true, tableRef.Columns, values);
+                            displayableArmProperties.Add(displayTable);
+                            break;
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                string errorValue = $"Failed to get Arm Response, Error: {e.Message}";
+
+                foreach (IReferenceAlertProperty propertyRef in propertyRefs.OfType<IReferenceAlertProperty>())
+                {
+                    switch (propertyRef)
+                    {
+                        case TextReferenceAlertProperty textRef:
+                            TextAlertProperty displayText = new TextAlertProperty(textRef.PropertyName, textRef.DisplayName, textRef.Order, errorValue);
+                            displayableArmProperties.Add(displayText);
+                            break;
+
+                        case LongTextReferenceAlertProperty longTextRef:
+                            LongTextAlertProperty displayLongText = new LongTextAlertProperty(longTextRef.PropertyName, longTextRef.DisplayName, longTextRef.Order, errorValue);
+                            displayableArmProperties.Add(displayLongText);
+                            break;
+
+                        case KeyValueReferenceAlertProperty keyValueRef:
+                            IDictionary<string, string> keyValueField = new Dictionary<string, string> { { keyValueRef.ReferencePath, errorValue } };
+                            KeyValueAlertProperty displayKeyValue = new KeyValueAlertProperty(keyValueRef.PropertyName, keyValueRef.DisplayName, keyValueRef.Order, keyValueField);
+                            displayableArmProperties.Add(displayKeyValue);
+                            break;
+
+                        case TableReferenceAlertProperty tableRef:
+                            displayText = new TextAlertProperty(tableRef.PropertyName, tableRef.DisplayName, tableRef.Order, errorValue);
+                            displayableArmProperties.Add(displayText);
+                            break;
+                    }
+                }
```

Comment:
maybe extract this to a method

Reply:
I can extract the "non error" switch case into the same function if cleaner.
The code is the same except for the table. So i can add an error flag to the signature to differentiate.